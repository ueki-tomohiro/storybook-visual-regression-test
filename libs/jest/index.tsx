import { composeStories, StoryFile } from "@storybook/testing-react";
import { render } from "@testing-library/react";
import React from "react";

export const renderFromStories = (stories: StoryFile) => {
  const composed = composeStories(stories);
  const testCases = Object.values(composed).map((Story: any) => [Story.storyName, Story, Story.args]);
  test.each(testCases)("renders %s", (_storyName, Component: any, args) => {
    if (_storyName.match(/Fake/)) return;

    const tree = render(<Component {...(args || {})} />);
    expect(tree.baseElement).toMatchSnapshot();
  });
};
