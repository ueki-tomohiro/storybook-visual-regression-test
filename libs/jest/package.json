{
  "name": "@demo-libs/jest",
  "version": "0.0.1",
  "private": true,
  "description": "Demo Jest",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.21.4",
    "@babel/preset-env": "^7.21.4",
    "@babel/preset-typescript": "^7.21.4",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.5.1",
    "@types/jest": "^29.5.1",
    "babel-jest": "^29.5.0",
    "babel-plugin-transform-import-meta": "^2.2.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0"
  },
  "engines": {
    "node": "^20",
    "yarn": ">=1"
  },
  "volta": {
    "node": "20.0.0",
    "yarn": "1.22.19"
  }
}
