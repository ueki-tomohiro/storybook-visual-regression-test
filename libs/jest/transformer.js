const babelJest = require("babel-jest");
const babelOptions = {
  presets: [
    [
      "@babel/env",
      {
        targets: {
          node: "current",
        },
        exclude: ["transform-regenerator"],
      },
    ],
    "@babel/preset-typescript",
    "@babel/react",
  ],
  plugins: ["babel-plugin-transform-import-meta"],
  babelrc: false,
};
module.exports = babelJest.createTransformer(babelOptions);
