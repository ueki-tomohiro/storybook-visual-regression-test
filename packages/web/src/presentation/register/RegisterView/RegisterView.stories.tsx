import { action } from "@storybook/addon-actions";

import { RegisterView, RegisterViewProps } from "./RegisterView";

const defaultArgs: RegisterViewProps = {
  regisiterTodo: action("regisiterTodo"),
};

export default {
  title: "views/RegisterView",
  component: RegisterView,
};

export const Default = { args: defaultArgs };

export const Loading = { args: { isLoading: true } };
