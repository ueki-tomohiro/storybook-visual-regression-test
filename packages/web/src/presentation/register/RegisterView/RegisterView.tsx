import { TodoRegisterble } from "@demo/api/lib/demo/model";
import { Button } from "@demo/ui/components/Button";
import { InputField } from "@demo/ui/components/InputField";
import { yupResolver } from "@hookform/resolvers/yup";
import React from "react";
import { SubmitHandler, useForm } from "react-hook-form";
import * as Yup from "yup";

export type RegisterViewProps = {
  regisiterTodo: (args: TodoRegisterble) => void;
  isLoading?: boolean;
};

export const RegisterView: React.FC<RegisterViewProps> = ({ regisiterTodo, isLoading }) => {
  const resolver = yupResolver(
    Yup.object().shape({
      description: Yup.string().required("詳細を入力してください"),
      end_date: Yup.date().typeError("完了日を入力してください").required("完了日を入力してください"),
    })
  );
  const {
    handleSubmit,
    register,
    formState: { errors, isSubmitting },
  } = useForm<TodoRegisterble>({
    defaultValues: {
      description: "",
      end_date: "",
    },
    resolver,
  });

  const onSubmit: SubmitHandler<TodoRegisterble> = (values) => {
    regisiterTodo(values);
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <InputField
        {...register("description")}
        label="詳細"
        disabled={isSubmitting || isLoading}
        error={errors.description?.message}
      />
      <InputField
        {...register("end_date")}
        label="完了日"
        type="date"
        disabled={isSubmitting || isLoading}
        error={errors.end_date?.message}
      />
      <Button variant="primary" type="submit" disabled={isSubmitting || isLoading} isLoading={isLoading}>
        登録
      </Button>
    </form>
  );
};
