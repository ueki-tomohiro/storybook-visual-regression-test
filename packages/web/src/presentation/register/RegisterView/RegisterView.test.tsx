import { renderFromStories } from "@demo-libs/jest";

import * as stories from "./RegisterView.stories";

describe(stories.default.title, () => {
  renderFromStories(stories);
});
