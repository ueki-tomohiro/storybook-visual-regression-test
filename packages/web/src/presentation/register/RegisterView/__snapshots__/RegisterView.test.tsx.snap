// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/RegisterView renders Default 1`] = `
<body>
  <div>
    <form>
      <div
        class="relative w-full"
      >
        <fieldset
          class="w-full h-12 mb-6 rounded border border-solid border-tertiary focus-within:border-secondary [&_label]:[&:not(:focus-within)]:text-tertiary [&_label]:focus-within:text-secondary"
        >
          <legend
            class="min-h-3 ml-2 flex px-0.5"
          >
            <label
              class="flex text-xs"
            >
              詳細
            </label>
          </legend>
        </fieldset>
      </div>
      <div
        class="relative w-full"
      >
        <fieldset
          class="w-full h-12 mb-6 rounded border border-solid border-tertiary focus-within:border-secondary [&_label]:[&:not(:focus-within)]:text-tertiary [&_label]:focus-within:text-secondary"
        >
          <legend
            class="min-h-3 ml-2 flex px-0.5"
          >
            <label
              class="flex text-xs"
            >
              完了日
            </label>
          </legend>
        </fieldset>
      </div>
      <button
        class="mx-auto max-w-3xl text-white bg-primary disabled:bg-tertiary rounded px-4 py-2"
        type="submit"
      >
        登録
      </button>
    </form>
  </div>
</body>
`;

exports[`views/RegisterView renders Loading 1`] = `
<body>
  <div>
    <form>
      <div
        class="relative w-full"
      >
        <fieldset
          class="w-full h-12 mb-6 rounded border border-solid border-tertiary focus-within:border-secondary [&_label]:[&:not(:focus-within)]:text-tertiary [&_label]:focus-within:text-secondary"
        >
          <legend
            class="min-h-3 ml-2 flex px-0.5"
          >
            <label
              class="flex text-xs"
            >
              詳細
            </label>
          </legend>
        </fieldset>
      </div>
      <div
        class="relative w-full"
      >
        <fieldset
          class="w-full h-12 mb-6 rounded border border-solid border-tertiary focus-within:border-secondary [&_label]:[&:not(:focus-within)]:text-tertiary [&_label]:focus-within:text-secondary"
        >
          <legend
            class="min-h-3 ml-2 flex px-0.5"
          >
            <label
              class="flex text-xs"
            >
              完了日
            </label>
          </legend>
        </fieldset>
      </div>
      <button
        class="mx-auto max-w-3xl text-white bg-primary disabled:bg-tertiary rounded px-4 py-2"
        disabled=""
        type="submit"
      >
        <span
          class="material-symbols-outlined animate-rotate"
        >
          refresh
        </span>
      </button>
    </form>
  </div>
</body>
`;
